pandas/tests/series/indexing/test_setitem.py

```diff
@@ -210,6 +210,14 @@ def test_setitem_nan_casts(self):
         ser[:5] = np.nan
         tm.assert_series_equal(ser, expected)
 
+    @pytest.mark.parametrize("dtype", ["int64", "Int64"])
+    def test_setitem_assigning_different_dtype(self, dtype):
+        # GH: 20635
+        ser = Series(["3", "4"], name="A")
+        ser.loc[:] = ser.loc[:].astype("int64").astype(dtype)
```

Comment:
the `ser.loc[:]` on the RHS should be unnecessary shouldnt it?

Reply:
Should not matter I think, wanted to use the same as in the issue

Reply:
if its not necessary to the behavior being tested, go with the simpler version

Reply:
Thx. Simplified the test accordingly